deployment/terraform/agent.go

```diff
@@ -0,0 +1,140 @@
+package terraform
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-load-test-ng/coordinator"
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/agent"
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/cluster"
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/performance"
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/performance/prometheus"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform/ssh"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+func (t *Terraform) generateLoadtestAgentConfig(output *terraformOutput) loadtest.Config {
+	return loadtest.Config{
+		ConnectionConfiguration: loadtest.ConnectionConfiguration{
+			// TODO: replace with reverse nginx ip
+			ServerURL:                   "http://" + output.Proxy.Value.PrivateIP,
+			WebSocketURL:                "ws://" + output.Proxy.Value.PrivateIP,
+			AdminEmail:                  t.config.AdminEmail,
+			AdminPassword:               t.config.AdminPassword,
+			IdleConnTimeoutMilliseconds: 90000,
+		},
+		UserControllerConfiguration: loadtest.UserControllerConfiguration{
+			Type: "simple",
+			Rate: 1.0,
+		},
+		InstanceConfiguration: loadtest.InstanceConfiguration{
+			NumTeams: 2,
+		},
+		UsersConfiguration: loadtest.UsersConfiguration{
+			InitialActiveUsers: 4,
+			MaxActiveUsers:     1000,
+		},
+	}
+}
+
+func (t *Terraform) startCoordinator(extAgent *ssh.ExtAgent, ip string) error {
+	sshc, err := extAgent.NewClient(ip)
+	if err != nil {
+		return err
+	}
+
+	// Populate the DB.
+	mlog.Info("Populating DB")
+	cmd := fmt.Sprintf("cd mattermost-load-test-ng-%s && export PATH=$PATH:/usr/local/go/bin && go run ./cmd/loadtest init", t.config.SourceCodeRef)
+	if err := sshc.RunCommand(cmd); err != nil {
+		return err
+	}
+
+	// Starting coordinator.
+	mlog.Info("Starting coordinator", mlog.String("ip", ip))
+	cmd = fmt.Sprintf("cd mattermost-load-test-ng-%s && export PATH=$PATH:/usr/local/go/bin && go run ./cmd/coordinator", t.config.SourceCodeRef)
```

Comment:
Assuming that the master branch thing is taken care of, we can write it like this:
`cd mattermost-load-test-ng && PATH=$PATH:/usr/local/go/bin go run ./cmd/coordinator &`.